mypy/main.py

```diff
@@ -205,6 +207,39 @@ def invert_flag_name(flag: str) -> str:
     return '--no-{}'.format(flag[2:])
 
 
+if sys.platform == 'win32':
+    python_executable_prefix = 'py -{}'
+else:
+    python_executable_prefix = 'python{}'
+
+
+def _python_version_from_executable(python_executable: str) -> Tuple[int, int]:
+    try:
+        check = subprocess.check_output([python_executable, '-c',
+                                         'import sys; print(repr(sys.version_info[:2]))'],
+                                        stderr=subprocess.STDOUT).decode()
+    except subprocess.CalledProcessError:
+        return sys.version_info[:2]
+    else:
+        return ast.literal_eval(check)
+
+
+def _python_executable_from_version(python_version: Tuple[int, int]) -> Optional[str]:
+    str_ver = '.'.join(map(str, python_version))
+    try:
+        sys_exe = subprocess.check_output(python_executable_prefix.format(str_ver).split(' ') +
+                                          ['-c', 'import sys; print(sys.executable)'],
+                                          stderr=subprocess.STDOUT).decode().strip()
+    except subprocess.CalledProcessError:
+        return None
+    else:
+        # don't set if its the sys.executable as this will break things later
+        if sys_exe.lower() != sys.executable:
```

Comment:
Why the `lower()`. Case sensitive filepaths are a thing, so different cases could be meaningful here.

Reply:
`sys.executable` is lowercase on Windows, and I believe should be everywhere. I can also guard this to win32.

Reply:
Either:

* `sys.executable` is always lowercase, this isn't useful
* The subprocess and main process disagree on the casing of `sys.executable` - this is a pretty big red flag that they're _not_ the same executable, since the same python code is producing different results!

So this `.lower()` is at best useless and at most harmful

Reply:
Fair point. Its kinda strange actually, that it would be reported as lowercase. I will remove this all the same.